completor/main.py

```diff
@@ -179,116 +179,69 @@ def create(
-                        continue  # not an active well
-                    chunk, after_content_line_number = process_content(line_number, clean_lines_map)
-                    schedule_data = read_schedule.set_welsegs(schedule_data, active_wells, chunk)
-                    line_number = after_content_line_number + 1
-                    continue
-
-                elif keyword == Keywords.COMPLETION_SEGMENTS:
-                    if well_name not in list(active_wells):
-                        output_text += format_text(keyword, "")
-                        line_number += 1
-                        continue  # not an active well
-                    chunk, after_content_line_number = process_content(line_number, clean_lines_map)
-
-                    schedule_data = read_schedule.set_compsegs(schedule_data, active_wells, chunk)
-
-                    line_number = after_content_line_number + 1
-                    case.check_input(well_name, schedule_data)
-
-                well_names.append(well_name)
-
-            else:
-                progress_bar.update(len(lines) - prev_line_number)
-
-        output_text += "\n" + format_header(paths)
-        for i, well_name_ in tqdm(enumerate(well_names), total=len(well_names)):
-            logger.debug("Writing new MSW info for well %s", well_name_)
-            well = Well(well_name_, i, case, schedule_data[well_name_])
-            output = create_output.format_output(well, case, figure_name, paths)
-            output_text += "\n" + output
+        # Find the old data for each of the four main keywords.
+        for chunk in find_keyword_data(Keywords.WELL_SPECIFICATION, schedule):
+            clean_data = clean_raw_data(chunk, Keywords.WELL_SPECIFICATION)
+            meaningful_data = read_schedule.set_welspecs(meaningful_data, clean_data)
+
+        for chunk in find_keyword_data(Keywords.COMPLETION_DATA, schedule):
+            clean_data = clean_raw_data(chunk, Keywords.COMPLETION_DATA)
+            meaningful_data = read_schedule.set_compdat(meaningful_data, clean_data)
+
+        for chunk in find_keyword_data(Keywords.WELL_SEGMENTS, schedule):
+            clean_data = clean_raw_data(chunk, Keywords.WELL_SEGMENTS)
+            meaningful_data = read_schedule.set_welsegs(meaningful_data, clean_data)
+
+        for chunk in find_keyword_data(Keywords.COMPLETION_SEGMENTS, schedule):
+            clean_data = clean_raw_data(chunk, Keywords.COMPLETION_SEGMENTS)
+            meaningful_data = read_schedule.set_compsegs(meaningful_data, clean_data)
+
+        for i, well_name in tqdm(enumerate(active_wells.tolist()), total=len(active_wells)):
+            well = Well(well_name, i, case, meaningful_data[well_name])
+            compdat, welsegs, compsegs, bonus = create_output.format_output(well, case, figure_name)
+
+            for keyword in [Keywords.COMPLETION_SEGMENTS, Keywords.WELL_SEGMENTS, Keywords.COMPLETION_DATA]:
+                # TODO: Move to utils? parsing?
```

Comment:
I guess this one makes it fail in the CI?

Reply:
Yes! I forgot to remove it!

Reply:
Yes! Removing, it's already done, just forgot about the actual TODO: